src/use_query_client.rs

```diff
@@ -131,6 +139,46 @@ where
 
             value.borrow_mut().set_to_loaded();
         }
+        // Handle revalidation if we receive an interval that is different from the current one.
```

Comment:
In what situation would the interval be any different if its initialized with the user-defined value and never changed again afaik?

Reply:
So this handles a case where I use the same query in two places but with different refetch intervals set. Agreed that this is a bit of an odd edge case to account for; I followed the behavior of [Tanstack Query here](https://github.com/TanStack/query/blob/065b5f4317838ddb6257027c4aafa7a422b0c2ee/packages/query-core/src/queryObserver.ts#L222) to reduce ambiguity, but I can see how either way (updating the interval or keeping it the same) could both make sense.